### ISSUE TITLE
Prompt for passphrase not shown during merge conflict when commit signing is enabled

### ISSUE THREAD

**Issue by solvedbiscuit71** [jesseduffield/lazygit#4067]
**Reproduce the bug**

1. Create a directory `origin` and initialize it as bare repository
2. Clone this repo into two folder namely `alice` and `bob`
3. Following instructions are performed in the given order

```
alice # commit A and push to origin
bob # pull from origin
bob # commit B but don't push to origin
alice # commit C and push to origin
bob # push to origin, failed due to local branch diverged
bob # pull from origin, leads to merge conflict
bob # resolve conflict via merge
bob # open merge options `m` and selects `c`
```

The commit A, B, C doesn't matter. Just ensure that B and C will results in a merge conflict. On selecting `c` in merge options the interface seems unchanged, no prompt for passphrase and pressing enter again shows the following error.

> TLDR; The prompt for passphrase doesn't shown during a merge conflict (because the local branch diverged from the upstream) when commit signing is enabled. Note: while creating a normal commit (such as A, B, C) lazygit do prompt for passphrase.

![image](https://github.com/user-attachments/assets/3fe27534-5b73-4f5b-a209-bee2f01fc60f)

**What I Tried**

- If I disable commit signing, the merge commit is created.
- It's not a bug with git or ssh-agent, I tried running `git commit` and I got a prompted for passphrase.

**Version**

lazygit: 0.44.1 (homebrew, darwin, arm64) 
git: 2.47.0

**Comment by blackthorne7** [comment]
@alwyn does it show up below the keybind options at the bottom of lazyvim?

![Image](https://github.com/user-attachments/assets/872648c6-1987-4b99-b8a2-e4780f37ea46)
